src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilder.java

```diff
@@ -180,26 +200,125 @@ public void perform(
     // generate a kubeconfig for the cluster
     KubeConfig kubeConfig = KubeConfig.fromCluster(projectId, cluster);
 
-    // run kubectl apply
-    KubectlWrapper.runKubectlCommand(
+    JenkinsRunContext context =
         new JenkinsRunContext.Builder()
             .workspace(workspace)
             .launcher(launcher)
             .taskListener(listener)
             .run(run)
-            .build(),
+            .build();
+    KubectlWrapper kubectl = new KubectlWrapper(context, kubeConfig);
+    kubectl.runKubectlCommand(
+        context,
         kubeConfig,
         "apply",
-        ImmutableList.<String>of("-f", workspace.child(manifestPattern).getRemote()));
+        ImmutableList.<String>of("-f", workspace.child(manifestPattern).getRemote(), "-o", "yaml"));
+
+    if (verifyDeployments) {
+      verified = verify(kubectl, manifestPattern, workspace, listener.getLogger());
+      if (!verified) {
+        throw new AbortException("Kubernetes objects could not be verified.");
+      }
+    }
 
-    // run the after build step if it exists
-    // NOTE(craigatgoogle): Due to the reflective way this class is created, initializers aren't
-    // run, so we still have to check for null.
     if (afterBuildStep != null) {
       afterBuildStep.perform(kubeConfig, run, workspace, launcher, listener);
     }
   }
 
+  private Set<String> getKindsToVerify() {
+    return ImmutableSet.of("Deployment");
+  }
+
+  /*
+   * verify the application of the supplied manifest objects and return true if it succeeded
```

Comment:
Prefer use of consistent grammar for header docs. Also missing \@param, \@return, \@throws annotations.